src/eventing.ts

```diff
@@ -0,0 +1,170 @@
+    new TextFactory({ value: name })
+  ]
+
+  const body = new SourceSummaryFactory({ source })
+  return h.createContentResponse(title, [body])
+}
+
+function sourceTypeListing(clientID: string, factoryMetadata?: FactoryMetadata): ComponentFactory<any> {
+  const duckTypes: ClusterDuckType = ctx.dashboardClient.Get({
+    apiVersion: DiscoveryV1Alpha,
+    kind: DiscoveryV1AlphaClusterDuckType,
+    name: SourcesDuck,
+  })
+
+  return new ClusterDuckTypeListFactory({ duckTypes, factoryMetadata })
+}
+
+// TODO: Simplify logic/ extract functions/methods
+function sourceListing(clientID: string, factoryMetadata?: FactoryMetadata, sourceType?: string): ComponentFactory<any> {
+  const sourceDucks: ClusterDuckType = ctx.dashboardClient.Get({
+    apiVersion: DiscoveryV1Alpha,
+    kind: DiscoveryV1AlphaClusterDuckType,
+    name: SourcesDuck,
+  })
+  const { spec, status } = sourceDucks
+
+  const sourceMetas: ResourceMeta[] = spec.versions.reduce((acc: ResourceMeta[], cur) =>
+    acc.concat(status.ducks[cur.name]),
+    [])
+  sourceMetas.sort((a, b) => (a.kind.localeCompare(b.kind)))
+
+  const allSources: Source[] = sourceMetas.reduce((acc: Source[], cur) => {
+    if (!sourceType || cur.kind === sourceType) {
+      return acc.concat(ctx.dashboardClient.List({
+        apiVersion: cur.apiVersion,
+        kind: cur.kind,
+        namespace: ctx.namespace,
+      }))
+    }
+    return acc
+  }, [])
+
+  const sources: Source[] = allSources.reduce((acc: Source[], cur) =>
+    acc.find((s) => s.metadata.uid === cur.metadata.uid) ? acc : acc.concat(cur),
+    [])
+  sources.sort((a, b) => (a.metadata.name || '').localeCompare(b.metadata.name || ''));
+
+  return new SourceListFactory({ sources, factoryMetadata })
+}
+
+//TODO: Finish implementing the usage of `additional printer columns` passing in sources
+function typedSourceListing(params: ({ clientID: string, sourceType: string }), factoryMetadata?: FactoryMetadata): ComponentFactory<any> {
+  const { clientID, sourceType } = params
+
+  const ducks: ClusterDuckType = ctx.dashboardClient.Get({
+    apiVersion: DiscoveryV1Alpha,
+    kind: DiscoveryV1AlphaClusterDuckType,
+    name: SourcesDuck,
+  })
+  const { spec, status } = ducks
+
+  const sourceTypeMeta: ResourceMeta = spec.versions.reduce((acc: ResourceMeta, cur) => {
+    const meta = status.ducks[cur.name].find(d => d.kind === sourceType)
+    return meta ? meta : acc
+  }, { apiVersion: "", kind: "", scope: "" })
+  const group: string = sourceTypeMeta.apiVersion.split("/")[0]
+    // TODO: this is currently a guess and potentially risky
+  const plural: string = sourceTypeMeta.kind.toLocaleLowerCase()+'s'
+
+  const crd: V1CustomResourceDefinition = ctx.dashboardClient.Get({
+    apiVersion: 'apiextensions.k8s.io/v1',
+    kind: 'CustomResourceDefinition',
+    name: `${plural}.${group}`,
+  })
+
+  const latest = crd.spec.versions[crd.spec.versions.length - 1]
```

Comment:
It was supposed to be the latest version, but looking more into it, I don't know if the last version is guaranteed to be the latest. Also, maybe it would just be simpler to just use the `storedVersion` instead, for the sake of consistency.